src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java

```diff
@@ -166,6 +166,10 @@ private void prepareIndexCreator(final ApplicationContext context) {
 
 		if (context instanceof ConfigurableApplicationContext && indexCreator != null) {
 			((ConfigurableApplicationContext) context).addApplicationListener(indexCreator);
+			if (mappingContext instanceof CouchbaseMappingContext) {
```

Comment:
would it not make sense to take this opportunity to move the glueing of beans out of the main code path and into a @Configuration class?